File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
def solution(board, h, w):
    
    # dx = [-1, 0, 1, 0]
    # dy = [0, -1, 0, 1]
    dx = [0, 1, -1, 0]
    dy = [1, 0, 0, -1]
    length = len(board)
    answer = 0
    
    for i in range(4):
        nx, ny = w+dx[i], h+dy[i]

        if nx < 0 or nx >= length or ny < 0 or ny >= length:
            continue

        if board[h][w] == board[ny][nx]:
            answer += 1
    return answer

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.java
class Solution {
    public int solution(String[][] board, int h, int w) {
        int answer = 0;
        
        int[] dx = {1, -1, 0, 0};
        int[] dy = {0, 0, 1, -1};
        
        int len = board.length;
        
        for(int i=0; i<4; i++){
            int nx = h + dx[i];
            int ny = w + dy[i];
            
            if(nx < 0 || nx >= len || ny < 0 || ny >= len)
                continue;
            
            if(board[h][w].equals(board[nx][ny]))
                answer++;
        }
        
        return answer;
    }
}